source/extensions/omni.isaac.lab/omni/isaac/lab/envs/mdp/observations.py

```diff
@@ -231,6 +237,92 @@ def image(
     return images.clone()
 
 
+class image_features(ManagerTermBase):
+    """Extracted image features with a frozen encoder from images of a specific datatype from the camera sensor.
+
+    Calls :meth:`image` to get the images, then performs inference. On initialization,
+    for a model zoo different from the default, define model_zoo_cfg: A dictionary with string keys and callable values.
+    Should include "model", (mapped to a callable with no arguments to return the model), "preprocess" (mapped to
+    a callable which consumes the images and returns the preprocessed images),
+    and "inference" (mapped to a callable that provided the model, and the preproccessed images, returns the features.)
+    """
+
+    def __init__(
+        self,
+        cfg: ObservationTermCfg,
+        env: ManagerBasedEnv,
+        model_zoo_cfg: dict | None = None,
+        initialize_all: bool = False,
+        device: str = "cuda:0",
+    ):
+        super().__init__(cfg, env)
+        if model_zoo_cfg is None:
+            self.model_zoo_cfg = {
+                # For more info about Theia, see: https://theia.theaiinstitute.com/
+                "TheiaTiny": {
+                    "model": (
+                        lambda: AutoModel.from_pretrained(
+                            "theaiinstitute/theia-tiny-patch16-224-cddsv", trust_remote_code=True
+                        )
+                        .eval()
+                        .to(device)
+                    ),
+                    "preprocess": lambda img: (img - torch.amin(img, dim=(1, 2), keepdim=True)) / (
+                        torch.amax(img, dim=(1, 2), keepdim=True) - torch.amin(img, dim=(1, 2), keepdim=True)
+                    ),  # Rescale to [0, 1]
+                    "inference": lambda model, images: model.forward_feature(
+                        images, do_rescale=False, interpolate_pos_encoding=True
+                    ),
+                },
+                "ResNet18": {
+                    "model": lambda: models.resnet18(pretrained=True).eval().to(device),
+                    "preprocess": lambda img: (
+                        img.permute(0, 3, 1, 2)  # Convert [batch, height, width, 3] -> [batch, 3, height, width]
+                        # Normalize in the format expected by pytorch: https://pytorch.org/hub/pytorch_vision_resnet/
+                        - torch.tensor([0.485, 0.456, 0.406], device=img.device).view(1, 3, 1, 1)
+                    ) / torch.tensor([0.229, 0.224, 0.225], device=img.device).view(1, 3, 1, 1),
+                    "inference": lambda model, images: model(images),
+                },
+            }
+        self.reset_model(initialize_all=initialize_all)
+
+    # The following is named reset_model instead of reset as otherwise, it's called at the end of every episode
+    def reset_model(self, model_name: str | None = None, initialize_all: bool = False) -> None:
```

Comment:
Who is supposed to call this reset_model? If noone then this should be a private function. Please add docstrings.